generators/rest-adapter-core/src/main/java/uk/gov/justice/services/adapter/rest/filter/LoggerRequestDataFilter.java

```diff
@@ -145,6 +114,62 @@ private void addContentTypeAndAcceptIfPresent(final JsonObjectBuilder builder, f
         return Optional.empty();
     }
 
+    private Optional<String> idFromHeaderOrMetadataIfPresent(final MultivaluedMap<String, String> headers, final Optional<Metadata> payloadMetadata) {
+        return Optional.ofNullable(headers.getFirst(ID))
```

Comment:
could we pass this to a private method that takes a NAME and a lambda of the metadata -> metadata.id().toString() bit?

Would that look cleaner?